<a href="https://colab.research.google.com/github/hrmufid/Mini-Project-2/blob/main/Mini_Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Investigate Hotel Business using Data Visualization**

# Import the necessary library

In [5]:
# import libraries
import pandas as pd
import scipy
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.preprocessing import MinMaxScaler

# import shap
import warnings
warnings.filterwarnings('ignore')

In [2]:
# initialize random state
random_state = 42

# set plot style for whole notebook
sns.set_style('darkgrid')

# Load the dataset

In [3]:
# IMPORT DATA
data = pd.read_csv('https://raw.githubusercontent.com/hrmufid/Mini-Project-4/main/hotel_bookings_data.csv')

df = data.copy()
df.head()

hotel  is_canceled  lead_time  arrival_date_year arrival_date_month  \
0  Resort Hotel            0        342               2017          September   
1  Resort Hotel            0        737               2017          September   
2  Resort Hotel            0          7               2017          September   
3  Resort Hotel            0         13               2017          September   
4  Resort Hotel            0         14               2017          September   

   arrival_date_week_number  arrival_date_day_of_month  \
0                        27                          1   
1                        27                          1   
2                        27                          1   
3                        27                          1   
4                        27                          1   

   stays_in_weekend_nights  stays_in_weekdays_nights  adults  ...  \
0                        0                         0       2  ...   
1                        0                         0       2  ...   
2                        0                         1       1  ...   
3                        0                         1       1  ...   
4                        0                         2       2  ...   

   booking_changes  deposit_type  agent company days_in_waiting_list  \
0                3    No Deposit    NaN     NaN                    0   
1                4    No Deposit    NaN     NaN                    0   
2                0    No Deposit    NaN     NaN                    0   
3                0    No Deposit  304.0     NaN                    0   
4                0    No Deposit  240.0     NaN                    0   

  customer_type   adr  required_car_parking_spaces  total_of_special_requests  \
0      Personal   0.0                            0                          0   
1      Personal   0.0                            0                          0   
2      Personal  75.0                            0                          0   
3      Personal  75.0                            0                          0   
4      Personal  98.0                            0                          1   

   reservation_status  
0           Check-Out  
1           Check-Out  
2           Check-Out  
3           Check-Out  
4           Check-Out  

[5 rows x 29 columns]

# Context

Anda berperan sebagai anggota dari tim Data Scientist di sebuah perusahaan hotel. Sebagai anggota dari tim Data Scientist, Anda mendapatkan tanggung jawab untuk memberikan insight-insight yang berhubungan dengan performa bisnis hotel. Insight tersebut dapat dicari dengan eksplorasi data, seperti menganalisis bagaimana perilaku para pelanggan dalam memesan tiket hotel ataupun mencari faktor-faktor yang mempengaruhi pembatalan pemesanan tiket hotel. Kemudian sajikan insight yang Anda dapatkan menggunakan visualisasi dan data story telling.

#1. Data Preprocessing

Sebelum memulai pemrosesan data, tahap paling awal yang harus dilakukan adalah mempersiapkan data mentah menjadi data yang bersih dan siap diolah. Kamu harus melakukan data preprocessing, yakni menangani berbagai permasalahan data seperti data yang kosong, data yang tidak sesuai, hingga mengidentifikasi data-data yang tidak dibutuhkan.

##1.1 Descriptive Analysis

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 119390 entries, 0 to 119389
Data columns (total 29 columns):
 #   Column                          Non-Null Count   Dtype  
---  ------                          --------------   -----  
 0   hotel                           119390 non-null  object 
 1   is_canceled                     119390 non-null  int64  
 2   lead_time                       119390 non-null  int64  
 3   arrival_date_year               119390 non-null  int64  
 4   arrival_date_month              119390 non-null  object 
 5   arrival_date_week_number        119390 non-null  int64  
 6   arrival_date_day_of_month       119390 non-null  int64  
 7   stays_in_weekend_nights         119390 non-null  int64  
 8   stays_in_weekdays_nights        119390 non-null  int64  
 9   adults                          119390 non-null  int64  
 10  children                        119386 non-null  float64
 11  babies                          119390 non-null  int64  
 12  meal            

Dapat dilihat di dalam dataset terdapat 29 kolom dan 119390 baris (paling banyak) dan terdapat nilai null

In [6]:
df.isnull().sum()

hotel                                  0
is_canceled                            0
lead_time                              0
arrival_date_year                      0
arrival_date_month                     0
arrival_date_week_number               0
arrival_date_day_of_month              0
stays_in_weekend_nights                0
stays_in_weekdays_nights               0
adults                                 0
children                               4
babies                                 0
meal                                   0
city                                 488
market_segment                         0
distribution_channel                   0
is_repeated_guest                      0
previous_cancellations                 0
previous_bookings_not_canceled         0
booking_changes                        0
deposit_type                           0
agent                              16340
company                           112593
days_in_waiting_list                   0
customer_type                          0
adr                                    0
required_car_parking_spaces            0
total_of_special_requests              0
reservation_status                     0
dtype: int64

In [8]:
# cek apakah ada data customer yang duplicate
df.duplicated()

0         False
1         False
2         False
3         False
4         False
          ...  
119385    False
119386    False
119387    False
119388    False
119389    False
Length: 119390, dtype: bool

## 1.2 Statistical Analysis

In [9]:
df.describe()

is_canceled      lead_time  arrival_date_year  \
count  119390.000000  119390.000000      119390.000000   
mean        0.370416     104.011416        2018.156554   
std         0.482918     106.863097           0.707476   
min         0.000000       0.000000        2017.000000   
25%         0.000000      18.000000        2018.000000   
50%         0.000000      69.000000        2018.000000   
75%         1.000000     160.000000        2019.000000   
max         1.000000     737.000000        2019.000000   

       arrival_date_week_number  arrival_date_day_of_month  \
count             119390.000000              119390.000000   
mean                  27.165173                  15.798241   
std                   13.605138                   8.780829   
min                    1.000000                   1.000000   
25%                   16.000000                   8.000000   
50%                   28.000000                  16.000000   
75%                   38.000000                  23.000000   
max                   53.000000                  31.000000   

       stays_in_weekend_nights  stays_in_weekdays_nights         adults  \
count            119390.000000             119390.000000  119390.000000   
mean                  0.927599                  2.500302       1.856403   
std                   0.998613                  1.908286       0.579261   
min                   0.000000                  0.000000       0.000000   
25%                   0.000000                  1.000000       2.000000   
50%                   1.000000                  2.000000       2.000000   
75%                   2.000000                  3.000000       2.000000   
max                  19.000000                 50.000000      55.000000   

            children         babies  is_repeated_guest  \
count  119386.000000  119390.000000      119390.000000   
mean        0.103890       0.007949           0.031912   
std         0.398561       0.097436           0.175767   
min         0.000000       0.000000           0.000000   
25%         0.000000       0.000000           0.000000   
50%         0.000000       0.000000           0.000000   
75%         0.000000       0.000000           0.000000   
max        10.000000      10.000000           1.000000   

       previous_cancellations  previous_bookings_not_canceled  \
count           119390.000000                   119390.000000   
mean                 0.087118                        0.137097   
std                  0.844336                        1.497437   
min                  0.000000                        0.000000   
25%                  0.000000                        0.000000   
50%                  0.000000                        0.000000   
75%                  0.000000                        0.000000   
max                 26.000000                       72.000000   

       booking_changes          agent      company  days_in_waiting_list  \
count    119390.000000  103050.000000  6797.000000         119390.000000   
mean          0.221124      86.693382   189.266735              2.321149   
std           0.652306     110.774548   131.655015             17.594721   
min           0.000000       1.000000     6.000000              0.000000   
25%           0.000000       9.000000    62.000000              0.000000   
50%           0.000000      14.000000   179.000000              0.000000   
75%           0.000000     229.000000   270.000000              0.000000   
max          21.000000     535.000000   543.000000            391.000000   

                 adr  required_car_parking_spaces  total_of_special_requests  
count  119390.000000                119390.000000              119390.000000  
mean      101.831122                     0.062518                   0.571363  
std        50.535790                     0.245291                   0.792798  
min        -6.380000                     0.000000                   0.000000  
25%        69.290000                     0.000000             

Kemungkian terdapat outlier dilihat dari Q1, Q3 dan min max dari setiap kolom.


In [11]:
# Cek outliers
fig, axs = plt.subplot(9,1 dpi=95, figsize=(7,17))
i = 0
for col in df.columns:
  axs[i].boxplot(df[col], vert=False)
  axs[i].set_ylabel(col)
  i+-1
plt.show()

SyntaxError: invalid syntax. Perhaps you forgot a comma? (ipython-input-3590827483.py, line 2)